### PR TITLE
Add a basic XPlatProvider implementation for custom variable support

### PR DESCRIPTION
**Pull request by JasonVMo**
This adds an XPlatProvider implementation to the react-platform-adapter package.

- The provider takes a `PartialTheme` as the input parameter as well as the usual children.
- In native, the theme will be transformed into the right format and be routed to the provider in RSD
- In web the provider will be a no-op and just pass through to children
- I also added a `suppressCssVariableInsertion` flag that can be used to fork behavior in the `fluent-provider` package.
- This also bumps `react-strict-dom` to version 0.0.3 and bumps the TS version to 5.0.0 so the stylex types can be handled correctly in build.

This is all a basic first step and is currently unused. I'll add the usage and changes in fluent-provider as a separate PR.

### CODE REVIEW

**Review by behowell — APPROVED** [review]
Looks good!

File: packages/date-time-utilities/src/dateGrid/getBoundedDateRange.ts [hunk 1]
```diff
@@ -8,11 +8,12 @@ import { compareDatePart } from '../dateMath/dateMath';
...
 export const getBoundedDateRange = (dateRange: Date[], minDate?: Date, maxDate?: Date): Date[] => {
   let boundedDateRange = [...dateRange];
+  const zeroVal: Number = Number(0);
   if (minDate) {
-    boundedDateRange = boundedDateRange.filter((date: Date) => compareDatePart(date, minDate as Date) >= 0);
+    boundedDateRange = boundedDateRange.filter((date: Date) => compareDatePart(date, minDate as Date) >= zeroVal);
   }
   if (maxDate) {
-    boundedDateRange = boundedDateRange.filter((date: Date) => compareDatePart(date, maxDate as Date) <= 0);
+    boundedDateRange = boundedDateRange.filter((date: Date) => compareDatePart(date, maxDate as Date) <= zeroVal);
```
Comment (behowell) [inline packages/date-time-utilities/src/dateGrid/getBoundedDateRange.ts:16]: Fixes like this probably should go in the master branch instead of xplat. That way we don't have to worry about merge conflicts in unrelated code when updating the branch.

Also, I think the fix here is to change the return type of `compareDatePart` to lower-case `number`, since it's returning the difference of two lower-case `number` values. I'm guessing it was a typo to use upper-case `Number`.

File: packages/react-components/react-platform-adapter/src/XPlatProvider/XPlatProvider.tsx [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+// css variables should be inserted as usual for web
+export const suppressCssVariableInsertion = false;
+
+/**
+ * XPlatProvider is effectively a no-op for web scenarios
+ */
+export const XPlatProvider: React.FunctionComponent<XPlatProviderProps> = props => {
+  const { children } = props;
+  return <>{children}</>;
+};
```
[the remark this answers is not in the document]
Comment (JasonVMo) [reply packages/react-components/react-platform-adapter/src/XPlatProvider/XPlatProvider.tsx:13]: Good suggestion, trying that now

File: packages/date-time-utilities/src/dateGrid/getBoundedDateRange.ts (hunk 1 above)
Comment (JasonVMo) [reply packages/date-time-utilities/src/dateGrid/getBoundedDateRange.ts]: I made this switch. For other typescript 5 changes I agree we can do them in master.
